### PR TITLE
Migrate Aura Contracts and Update bb-a-usd

### PR DESCRIPTION
**Pull request by sajanrajdev**
Relates to https://github.com/Badger-Finance/badger-strategies/issues/85
Tackles https://github.com/Badger-Finance/Badger-Project-Planning/issues/110

### Changes Introduces
- Updated the cvxCrvRewards contract according to Aura's internal migration
- Updated the bb_a_USD reward and swap path to the new version of the pool
- Adapted the migration test to successfully process the old bb-a-usd, sweep the new one from the old strat, migrate, transfer the new bb-a-usd to the strat and harvest to process it.
- Adapted the rest of the tests as needed

image

The `test_expected_aura_rewards_match_minted` fails as well here (same as for the rest of the aura strategies). @shuklaayush and I have already looked into this in the past and we haven't been able to figure out why is this. I will take a look again but I don't think it should be a blocker.

### CODE REVIEW

**Comment by GalloDaSballo** [comment]
Note that the the `balanceOfRewards` uses a generic loop, while harvest has hardcoded path.

This means that the balanceOfRewards can potentially offer more tokens that what harvest can process.

Not a blocker, but worth keeping in mind

**Review by GalloDaSballo — APPROVED** [review]
LGTM

All tests passing for me.
Screenshot 2022-12-18 at 00 35 04

Screenshot 2022-12-18 at 00 35 09

File: tests/upgrade/test_migrate.py [hunk 1]
```diff
@@ -34,16 +54,32 @@ def new_strategy(vault_proxy, proxyAdmin, deployer):
...
-    return strat_proxy
 
 
-def test_check_storage_integrity(strat_proxy, vault_proxy, new_strategy):
-    with brownie.reverts():
-        strat_proxy.minBbaUsdHarvest()
+def test_migrate(strat_proxy, vault_proxy, new_strategy, current_bb_a_usd, new_bb_a_usd, gov, want):
+    ## 1. Reduce "minBbaUsdHarvest" threshold on old strategy and Harvest
+    current_bbausd_bal = current_bb_a_usd.balanceOf(strat_proxy)
+    strat_proxy.setMinBbaUsdHarvest(current_bbausd_bal - 1)
```
Comment (shuklaayush) [inline tests/upgrade/test_migrate.py:69]: I think we should just set this to 0 in the actual migration script

**Review by shuklaayush — APPROVED** [review]
LG

File: tests/upgrade/test_migrate.py (hunk 1 above)
Comment (sajanrajdev) [reply tests/upgrade/test_migrate.py:69]: Sounds good to me

**Pull request merged**